apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -579,22 +590,79 @@ public long countBlobs(String containerName, ListContainerOptions options) {
       throw new UnsupportedOperationException();
    }
 
-   // copied from BaseBlobStore
+   // TODO: Move statics up to BaseBlobStore to reuse?
+   @com.google.inject.Inject(optional = true)
+   @Named("jclouds.mpu.parallel.degree")
+   @VisibleForTesting
+   private int numThreads = 16;
+
+   private ExecutorService jcloudsBigBlobUploadTreadPool = MoreExecutors.getExitingExecutorService(
+         new ThreadPoolExecutor(
+         numThreads,
+         numThreads,
+         0L,
+         TimeUnit.MILLISECONDS,
+         new ArrayBlockingQueue<Runnable>(numThreads),
+         new ThreadPoolExecutor.CallerRunsPolicy()), 0, TimeUnit.NANOSECONDS);
+
    @Beta
    protected String putMultipartBlob(String container, Blob blob, PutOptions overrides) {
-      List<MultipartPart> parts = Lists.newArrayList();
+      if (overrides.getCustomExecutor() != null) {
+         return putMultipartBlob(container, blob, overrides, overrides.getCustomExecutor());
+      } else {
+         return putMultipartBlob(container, blob, overrides, jcloudsBigBlobUploadTreadPool);
+      }
+   }
+
+   // copied from BaseBlobStore
```

Comment:
We need similar code in `BaseBlobStore` for other providers now that we change `PutOptions`.  This will also resolve [JCLOUDS-227](https://issues.apache.org/jira/browse/JCLOUDS-227).